### PR TITLE
Implemented Eco viewer example.

### PR DESCRIPTION
**Pull request by Britefury**
This PR adds a viewer designed to be invoked by Eco to display Eco document content using LSpace.

### CODE REVIEW

File: examples/eco_viewer.rs [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    fn pres(&self, style: &Rc<TextStyleParams>) -> Pres {
+        return Column::new(self.lines.iter().map(|x| x.pres(style)).collect());
+    }
+}
+
+
+fn main() {
+    // Initialise GTK
+    gtk::init().unwrap_or_else(|_| panic!("Failed to initialize GTK."));
+    println!("Major: {}, Minor: {}", gtk::get_major_version(), gtk::get_minor_version());
```
Comment (ltratt) [inline examples/eco_viewer.rs]: Do we need to print this out?

**Comment by Britefury** [comment]
The Eco viewer has been renamed to JSON viewer, as it now accepts a JSON description of a generic presentation, rather than a domain specific presentation (it was a document consisting of lines, that consist of tokens, each of which is a text item).

This allows the Eco side complete control over what gets displayed. The Eco side now provides an API that follows the Rust API quite closely, so other team members could get a taste of what is to come.

**Comment by ltratt** [comment]
This looks fine to me. We should wait until the person reviewing the Eco end of things has verified this works well for them, then both PRs can be merged somewhat-in-sync.

**Comment by ptersilie** [comment]
Small suggestion: Can we maybe change the font to a fixed-width one (e.g. Monospace)?

**Comment by Britefury** [comment]
Font changed, also gets input from STDIN; makes non-blocking operation easier from Python side.

**Comment by ltratt** [comment]
OK, so shall we do some squashing (not necessarily down to a single commit! but just getting rid of some of the unimportant commits) and merge? Sound sensible @ptersilie?

**Comment by Britefury** [comment]
Given the fairly haphazard order in which things were done, squashing down to one will be easier I think! :)

**Comment by Britefury** [comment]
Squashed.

**Pull request merged**